swagger.yaml

```
@@ -3173,6 +3173,253 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+
+  /v2/infra-envs/{infra_env_id}/hosts:
+    post:
+      tags:
+        - installer
+      security:
+        - agentAuth: []
+      description: Registers a new OpenShift agent.
+      operationId: v2RegisterHost
+      parameters:
+        - in: path
+          name: infra_env_id
+          description: The InfraEnv that the agent is associated with.
+          type: string
+          format: uuid
+          required: true
+        - in: body
+          name: new-host-params
+          description: The description of the agent being registered.
+          required: true
+          schema:
+            $ref: '#/definitions/host-create-params'
+        - in: header
+          name: discovery_agent_version
+          description: The software version of the discovery agent that is registering the agent.
+          type: string
+          required: false
+      responses:
+        "201":
+          description: Success.
+          schema:
+            $ref: '#/definitions/host_registration_response'
+        "400":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "401":
+          description: Unauthorized.
+          schema:
+            $ref: '#/definitions/infra_error'
+        "403":
+          description: Forbidden.
+          schema:
+            $ref: '#/definitions/infra_error'
+        "404":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        "405":
+          description: Method Not Allowed.
+          schema:
+            $ref: '#/definitions/error'
+        "409":
+          description: Cluster cannot accept new agents due to its current state.
```

Is this only due to the cluster's state (if a cluster-ID is passed) or also for the InfraEnv's state?


```suggestion
          description: Cluster cannot accept new hosts due to its current state.
```

---

Currently only cluster